R/derive_merged.R

```diff
@@ -1038,3 +1038,170 @@ derive_vars_merged_lookup <- function(dataset,
+#'
+#' @param filter_add Filter for additional dataset (`dataset_add`)
+#'
+#'   Only observations fulfilling the specified condition are taken into account
+#'   for summarizing. If the argument is not specified, all observations are
+#'   considered.
+#'
+#'   *Permitted Values*: a condition
+#'
+#' @param analysis_var Analysis variable
+#'
+#'   The values of the specified variable are summarized by the function
+#'   specified for `summary_fun`.
+#'
+#' @param summary_fun Summary function
+#'
+#'   The specified function that takes as input `analysis_var` and performs the
+#'   calculation. This can include built-in functions as well as user defined
+#'   functions, for example `mean` or `function(x) mean(x, na.rm = TRUE)`.
+#'
+#' @author Stefan Bundfuss
+#'
+#' @details
+#'
+#'   1. The records from the additional dataset (`dataset_add`) are restricted
+#'   to those matching the `filter_add` condition.
+#'
+#'   1. The values of the analysis variable (`analysis_var`) are summarized by
+#'   the summary function (`summary_fun`) for each by group (`by_vars`) in the
+#'   additional dataset (`dataset_add`).
+#'
+#'   1. The summarized values are merged to the input dataset as a new variable
+#'   (`new_var`). For observations without a matching observation in the
+#'   additional dataset the new variable is set to `NA`. Observations in the
+#'   additional dataset which have no matching observation in the input dataset
+#'   are ignored.
+#'
+#' @return The output dataset contains all observations and variables of the
+#'   input dataset and additionally the variable specified for `new_var`.
+#'
+#' @family der_gen
+#' @keywords der_gen
+#'
+#' @seealso [derive_summary_records()], [get_summary_records()]
+#'
+#' @export
+#'
+#' @examples
+#' library(tibble)
+#'
+#' # add a variable for the mean of AVAL within each visit
+#' adbds <- tribble(
+#'   ~AVISIT,  ~ASEQ, ~AVAL,
+#'   "WEEK 1",     1,    10,
+#'   "WEEK 1",     2,    NA,
+#'   "WEEK 2",     3,    NA,
+#'   "WEEK 3",     4,    42,
+#'   "WEEK 4",     5,    12,
+#'   "WEEK 4",     6,    12,
+#'   "WEEK 4",     7,    15
+#' )
+#'
+#' derive_var_merged_summary(
+#'   adbds,
+#'   dataset_add = adbds,
+#'   by_vars = vars(AVISIT),
+#'   new_var = MEANVIS,
+#'   analysis_var = AVAL,
+#'   summary_fun = function(x) mean(x, na.rm = TRUE)
+#' )
+#'
+#' # add a variable listing the lesion ids at baseline
+#' adsl <- tribble(
+#'   ~USUBJID,
+#'   "1",
+#'   "2",
+#'   "3"
+#' )
+#'
+#' adtr <- tribble(
+#'   ~USUBJID, ~AVISIT,    ~LESIONID,
+#'   "1",      "BASELINE", "INV-T1",
+#'   "1",      "BASELINE", "INV-T2",
+#'   "1",      "BASELINE", "INV-T3",
+#'   "1",      "BASELINE", "INV-T4",
+#'   "1",      "WEEK 1",   "INV-T1",
+#'   "1",      "WEEK 1",   "INV-T2",
+#'   "1",      "WEEK 1",   "INV-T4",
+#'   "2",      "BASELINE", "INV-T1",
+#'   "2",      "BASELINE", "INV-T2",
+#'   "2",      "BASELINE", "INV-T3",
+#'   "2",      "WEEK 1",   "INV-T1",
+#'   "2",      "WEEK 1",   "INV-N1"
+#' )
```

Comment:
Can we add in dummy SUBJIDs since we have the real estate?

Reply:
I am not sure what you mean. Do you want different values for `USUBJID` than `"1"`, and `"2"`?